src/library/dao/trackdao.cpp

```diff
@@ -1737,14 +1737,34 @@ bool TrackDAO::updateTrack(const Track& track) const {
     return true;
 }
 
+// Make sure that `directory` in in track_locations table is indeed a
+// directory path. This works around / removes residues of a bug where tracks
+// are falsely marked missing because `directory` == `location`.
+void TrackDAO::cleanupTrackLocationsDirectory() const {
+    kLogger.debug() << "cleanupTrackLocationsDirectory"
+                    << QThread::currentThread() << m_database.connectionName();
+    QSqlQuery query(m_database);
+    // If directory == location, we remove the filename and the trailing /
+    // As of now, this is the only inconsistency we know of. We might also verify
+    // directory (eg. no triling /) or recreate file name and directory from
```

Comment:
```suggestion
    // directory (eg. no trailing /) or recreate file name and directory from
```

Reply:
Thanks, interesting codespell missed that one.
(I'm not responsible, I wrote this outdoors and my screen has just 400nits ; )